pkg/chunk/gcp/bigtable_index_client.go

```diff
@@ -180,6 +198,9 @@ func (s *storageClientColumnKey) BatchWrite(ctx context.Context, batch chunk.Wri
 			muts = append(muts, mut)
 		}
 
+		s.writeLimiter.WaitN(ctx, len(muts))
```

Comment:
Given we're using the `WaitN()`, does the `ctx` have a deadline set? If not, it may be risky cause it could potentially hung for a long time, while a fast failing may be more desiderable.

Reply:
I have a default deadline is not set.